src/Module/AllFriends.php

```diff
@@ -68,13 +71,13 @@ public static function content()
 			} else {
 				$connlnk = $app->getBaseURL() . '/follow/?url=' . $friend['url'];
 				$photoMenu = [
-					'profile' => [L10n::t("View Profile"), Model\Contact::magicLink($friend['url'])],
-					'follow'  => [L10n::t("Connect/Follow"), $connlnk]
+					'profile' => [L10n::t('View Profile'), Model\Contact::magicLinkbyId($friend['id'])],
+					'follow'  => [L10n::t('Connect/Follow'), $connlnk]
 				];
 			}
 
 			$entry = [
-				'url'          => Model\Contact::magicLink($friend['url']),
+				'url'          => Model\Contact::magicLinkbyId($friend['id']),
```

Comment:
It looks like `magicLinkbyId` takes a second parameter with the contact URL. Since we have it in `$friend`, could you please use it?